### PR TITLE
Support Spark math functions for asinh, acosh, atanh, sec, csc

### LINKED ISSUE #4920: Support Spark asinh, acosh, atanh, sec, csc math functions.
### Description

Spark support [asinh](https://spark.apache.org/docs/latest/api/sql/#asinh), [acosh](https://spark.apache.org/docs/latest/api/sql/#acosh), [atanh](https://spark.apache.org/docs/latest/api/sql/#atanh) since 3.0.0.

Spark support [csc](https://spark.apache.org/docs/latest/api/sql/#csc), [sec](https://spark.apache.org/docs/latest/api/sql/#sec) since 3.3.0.

### PR DESCRIPTION
**Pull request by Yohahaha**
Support Spark math functions for asinh, acosh, atanh.

Fixed #4920

### CODE REVIEW

**Comment by Yohahaha** [comment]
Hi @mbasmanova could you help review this PR?

**Review by mbasmanova — COMMENTED** [review]
@bikramSingh91 Bikram, would you help review this PR?

File: velox/functions/sparksql/tests/ArithmeticTest.cpp [hunk 1]
```diff
@@ -183,6 +183,64 @@ TEST_F(ArithmeticTest, Divide) {
   EXPECT_TRUE(std::isnan(divide(kInf, -kInf).value_or(0)));
 }
 
+TEST_F(ArithmeticTest, acosh) {
```
Comment (bikramSingh91) [inline velox/functions/sparksql/tests/ArithmeticTest.cpp:186]: nit: wondering if it makes sense to add only one case each to these tests that has a valid value and a valid result

**Review by bikramSingh91 — APPROVED** [review]
Looks good, just a small nit

File: velox/functions/sparksql/tests/ArithmeticTest.cpp (hunk 1 above)
Comment (Yohahaha) [reply velox/functions/sparksql/tests/ArithmeticTest.cpp:186]: Corner cases should enough, other cases may have precision problem on different compiler.

**Pull request closed without merging**